### ISSUE TITLE
utteranceQueue/Utterance could manage "First Time" alerts

### ISSUE THREAD

**Issue by zepumph** [phetsims/scenery-phet#406]
Often in designing alerts, we have places where an alert changes verbosity after N (usually 1) time(s) times being alerted. For example in friction when moving the book down:

First two times: `Down. Rub fast or slow.`
Subsequent times: `Down.`

What if we could register this with `Utterence` like 
new Utterance( {
  initialAlert: 'Down. Rub fast or slow.'
  subsequentAlert: 'Down'.
  numberOfInitialAlerts: 2
});

And then be able to call `Utterance.reset()` when the reset button is called.

Would others have a use for this? Any other ideas  or suggestions?

**Comment by zepumph** [comment]
The best part about this to me seems like the ability to still just be able to pass that Utterance into utteranceQueue, like `utteranceQueue.addToBack( myUtterance)` and the Utterance itself is the state data, and `utteranceQueue.next()` would then hold the logic to parse the Utterance to get the appropriate string.

**Comment by jessegreenberg** [comment]
Yes, this seems very useful for Utterance. I thought for a moment about whether this should be implemented in a subtype of Utterance, but having to require different kinds of Utterances sounds like a bummer and this seems pretty light weight. Good idea!

**Comment by zepumph** [comment]
Hmmm, if we are looking for the general case, maybe we should have an array of possible strings, instead of just supporting a first and "after" alert.

For example from this request:
>After two number 8 alerts, "Jiggling very fast, very hot." you get one "Reset" hint alert, and then again, after two more number 8 alerts, you get another "Reset" hint alert. Then you don't get any more until the sim is reset.

Basically the counter would "mod" by the array length and just repeat that. This is a different use case, not sure if they deserve separate options (as in both may be needed). Oh I've got it! All could be solved with an option of type string array, `alertSchema` and then a toggling option like `loopSchema`.

```js
var looper = new Utterance( {
alertSchema: [ 'do number 1', 'do number 2', 'do number 3'],
loopSchema: true
});


var repeatLast = new Utterance( {
alertSchema: [ 'do number 1', 'do number 2', 'do number 3'],
loopSchema: false
});
```

`looper would continue looping through the alertSchema list, in effect modding the number of times alerted by the array length, where `repeatLast` would just take the `Math.min` of the array's length and the number of times the Utterance has been alerted.

**Comment by zepumph** [comment]
>Hmmm, if we are looking for the general case, maybe we should have an array of possible strings, instead of just supporting a first and "after" alert.

BTW in friction, I have implemented both of these manually, see `BookMovementDescriber.numberOfTimesAlertedAtBottom`, and `TemperatureIncreasingDescriber.alertIncrease()` (indexing the INCREASE const)

**Comment by zepumph** [comment]
In the above commits I:

* seriously reworked Utterance, changing the signature to a `config` with a required 'alert' param.
* This alert param can be a string, or an array of strings.
* I update usages of `new Utterance` to account for that. I think I got all of them, especially because Utterance is not super to any classes.
* I then added support for the alert as an array. Currently there is only the originally intended feature from this issue: Utterance will keep track of when it is alerted, and get the next string from the array each time. When it reaches the end, it will repeat the last alert until it is reset.
* I then spent a fair bit of time adding UtterenceQueue tests, which include a backdoor in AriaHerald so that we can get at the alerts as downstream as possible, and make sure that they look like they should.

@jessegreenberg please review and make recommendations.

**Comment by jessegreenberg** [comment]
@zepumph changes to Utterance look really great. You also converted to es6, which is so nice. While reviewing UtteranceTests, I found a bug that I introduced in #412, thanks for making these.

**Comment by jessegreenberg** [comment]
Just a couple of questions about the tests. Would it be helpful if AriaHerald, used an emitter to signal when it announcedPolite instead? Would that be bettern than using _testingBackDoorCallback?

I am not sure I understand how testOrder functions are working in the tests, consider
```js
    var alert = new Utterance( {
      alert: [ '1', '2', '3' ]
    } );
    let alert4 = () => {
      utteranceQueue.addToBack( alert );
      utteranceQueue.addToBack( alert );
      utteranceQueue.addToBack( alert );
      utteranceQueue.addToBack( alert );
    };
    let testOrder = ( messageSuffix ) => {

      // newest at lowest index
      assert.ok( alerts[ 3 ] === '1', 'Array order1' + messageSuffix );
      assert.ok( alerts[ 2 ] === '2', 'Array order2' + messageSuffix );
      assert.ok( alerts[ 1 ] === '3', 'Array order3' + messageSuffix );
      assert.ok( alerts[ 0 ] === '3', 'Array order4' + messageSuffix );
    };

    alert4();
    await timeout( sleepTiming * 4 );
    testOrder( '' );
```
I would expect alerts array to look like [1, 2, 3, 3], but testOrder looks like it is verifying the opposite. Is this an issue with the current implementation or the test?

**Comment by jessegreenberg** [comment]
Is it because of the unshift here?
```js
      // keep track of all alerts being put on aria live elements so we can test against them.
      AriaHerald.initialize( text => {
        alerts.unshift( text );
      } );
```

**Comment by zepumph** [comment]
I'm going to do a few updates to AriaHerald. I'm going to make it a class, convert it to es6, give it an emitter, and see about integrating that into the tests. I'm going to move this into a new issue though.

**Comment by zepumph** [comment]
@jessegreenberg much of the above code changed as part of #414. Is there anything else to do here?

**Comment by jessegreenberg** [comment]
Thanks @zepumph, this can be closed and we can continue with the #414.
